modules/mobile-agents/android/client/plugins/IDPProxy/src/org/wso2/mdm/agent/proxy/utils/ServerUtilities.java

```diff
@@ -157,33 +174,62 @@ public static HttpRequestBase buildHeaders(HttpRequestBase httpRequestBase,
 			HttpResponse response = httpClient.execute(httpGetWithHeaders);
 			responseParams.put(Constants.SERVER_RESPONSE_BODY, getResponseBody(response));
 			responseParams.put(Constants.SERVER_RESPONSE_STATUS,
-			                   String.valueOf(response.getStatusLine().getStatusCode()));
+					String.valueOf(response.getStatusLine().getStatusCode()));
 		} catch (ClientProtocolException e) {
 			throw new IDPTokenManagerException("Invalid client protocol.", e);
 		} catch (IOException e) {
 			responseParams.put(Constants.SERVER_RESPONSE_BODY, "Internal Server Error");
 			responseParams.put(Constants.SERVER_RESPONSE_STATUS,
-			                   Constants.INTERNAL_SERVER_ERROR);
+					Constants.INTERNAL_SERVER_ERROR);
 			throw new IDPTokenManagerException("Server connectivity failure.", e);
 		}
-		
+
+		return responseParams;
+	}
+
+	public static Map<String, String> sendDeleteRequest(String url, String params, Map<String, String> headers)
+			throws IDPTokenManagerException {
+
+		Map<String, String> responseParams = new HashMap<String, String>();
+		HttpDelete httpDelete = new HttpDelete(url);
+		HttpDelete httpDeleteWithHeaders = (HttpDelete) buildHeaders(httpDelete, headers);
+
+		try {
+
+			HttpClient httpClient = getCertifiedHttpClient();
+
+			HttpResponse response = httpClient.execute(httpDeleteWithHeaders);
+			responseParams.put(Constants.SERVER_RESPONSE_BODY, getResponseBody(response));
+			responseParams.put(Constants.SERVER_RESPONSE_STATUS,
+					String.valueOf(response.getStatusLine().getStatusCode()));
+
+		} catch (ClientProtocolException e) {
+			throw new IDPTokenManagerException("Invalid client protocol.", e);
+		} catch (IOException e) {
+			responseParams.put(Constants.SERVER_RESPONSE_BODY, "Internal Server Error");
```

Comment:
Shouldn't we be setting these two parameters within the previous catch block as well?